appmap/django.py

```diff
@@ -140,16 +142,27 @@ def request_params(request):
     return values_dict(params.lists())
 
 def get_resolved_match(path_info, resolver):
-    # Look through the resolver's cache of reverse
-    # route mappings and find the regex that matches the current
-    # request. The route mappings are tuples of the form:
-    #   (format string, regex, args, kwargs)
-    regex = None
-    for _,values in resolver.reverse_dict.lists():
-        for _,regex,_,_ in values:
-            match = re.match(regex, path_info[1:])
-            if match:
-                return (regex, match)
+    resolver_match = resolver.resolve(path_info)
+    regex = resolver_match.route
+
+    # Sometimes the route returned by resolve is a regex, sometimes
+    # it's a url pattern. If it's a regex, it will either have a named
+    # group, or it will end with a '$'. If it's a url pattern, convert
```

Comment:
Is that a fact? I'm wary that it'd turn out to be another assumption that isn't always true. Is it possible to look at the type, does it return a typed regex or a textual representation?

Reply:
TBH, I'm not 100% sure it's a fact. I'm going to walk through the code, as well as add some tests, to try to figure it out.

`resolver_match.route` is a string. AFAIK, Python doesn't have a regex type.